src/Symfony/Bundle/WebProfilerBundle/Tests/Controller/RouterControllerTest.php

```diff
@@ -0,0 +1,66 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Bundle\WebProfilerBundle\Tests\Controller;
+
+use Symfony\Bundle\FrameworkBundle\Test\WebTestCase;
+use Symfony\Bundle\WebProfilerBundle\Controller\RouterController;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\DataCollector\RequestDataCollector;
+use Symfony\Component\Routing\Matcher\UrlMatcherInterface;
+use Symfony\Component\Routing\RequestContext;
+use Symfony\Component\Routing\Route;
+use Symfony\Component\Routing\RouteCollection;
+use Twig\Environment;
+
+class RouterControllerTest extends WebTestCase
+{
+    public function testFalseNegativeTrace()
+    {
+        $routes = new RouteCollection();
+        $routes->add('route1', new Route('/foo/bar:123/baz'));
+
+        $request = Request::create('https://example.com/foo/bar:123/baz');
+        $response = new Response;
+
+        $requestDataCollector = new RequestDataCollector();
+        $requestDataCollector->collect($request, $response);
+        $requestDataCollector->lateCollect();
+
+        $matcher = $this->createMock(UrlMatcherInterface::class);
+
+        $matcher
+            ->method('getContext')
+            ->willReturn($this->createMock(RequestContext::class));
+
+        $controller = $this->getMockBuilder(RouterController::class)
+            ->setConstructorArgs([
+                'profiler' => null,
+                'twig' => $this->createMock(Environment::class),
+                'matcher' => $matcher,
+                'routes' => $routes
+            ])
+            ->onlyMethods(['panelAction'])
+            ->getMock();
+
+        $method = new \ReflectionMethod(RouterController::class, 'getTraces');
```

Comment:
Can we run a functional test against the controller instead of relying on implementation details and check the rendered result instead?

Reply:
Done